File: Packages/Flip54Core/Sources/Flip54Core/Prescription.swift
public enum Prescription: Codable, Hashable, Sendable {
    case reps(exercise: Exercise, count: Int)
    case hold(exercise: Exercise, seconds: Int)

    public var exercise: Exercise {
        switch self {
        case .reps(let exercise, _): return exercise
        case .hold(let exercise, _): return exercise
        }
    }

    public var isHold: Bool {
        if case .hold = self { return true }
        return false
    }

    public var displayString: String {
        switch self {
        case .reps(let exercise, let count):
            return "\(exercise.displayName) · \(count) reps"
        case .hold(let exercise, let seconds):
            let mins = seconds / 60
            let secs = seconds % 60
            let secsStr = secs < 10 ? "0\(secs)" : "\(secs)"
            let timeStr = mins > 0 ? "\(mins):\(secsStr)" : "\(secs)s"
            return "\(exercise.displayName) · Hold for \(timeStr)"
        }
    }
}

File: android/core/src/main/kotlin/com/flip54/core/Prescription.kt
package com.flip54.core

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
sealed class Prescription {
    @Serializable
    @SerialName("reps")
    data class Reps(val exercise: Exercise, val count: Int) : Prescription()

    @Serializable
    @SerialName("hold")
    data class Hold(val exercise: Exercise, val seconds: Int) : Prescription()

    val exercise: Exercise get() = when (this) {
        is Reps -> exercise
        is Hold -> exercise
    }

    val isHold: Boolean get() = this is Hold

    val displayString: String get() = when (this) {
        is Reps -> "${exercise.displayName} · $count reps"
        is Hold -> {
            val mins = seconds / 60
            val secs = seconds % 60
            val secsStr = if (secs < 10) "0$secs" else "$secs"
            val timeStr = if (mins > 0) "$mins:$secsStr" else "${secs}s"
            "${exercise.displayName} · Hold for $timeStr"
        }
    }
}
